core/src/block_sync.rs

```diff
@@ -44,6 +50,8 @@ pub struct BlockSynchronizer {
     gossip_size: NonZeroU32,
     network: IrohaNetwork,
     state: Arc<State>,
+    seen_blocks: BTreeSet<(NonZeroU64, HashOf<SignedBlock>)>,
+    latest_height: NonZeroU64,
```

Comment:
```suggestion
    seen_blocks: BTreeSet<(NonZeroUsize, HashOf<SignedBlock>)>,
    latest_height: NonZeroUsize,
```

Reply:
`latest_height` can be 0. That was a mistake

Reply:
The seen_blocks will be NonZeroU64 to match the block header data structure.